### DISCUSSION TITLE
Graph can not be compressed

### DISCUSSION THREAD

**Discussion by Siriusitzy** [deepmodeling/deepmd-kit#2692] [Q&A]
Hello! I encountered a model compression problem and hope to seek help from relevant people.

After I trained using deepmd/2.0-cuda11.3, the train.log showed 

>  Use tf.compat.v1.graph_util.extract_sub_graph
>  DEEPMD INFO 10054 ops in the final graph.

Then I used "**dp compress -i frozen_model.pb -o graph-compress.pb -l compress.log**"
No matter if I used deepmd/2.0-cuda11.3, deepmd/2.1, or deepmd/2.2, I got:

> DEEPMD INFO initialize model from scratch
> Traceback (most recent call last):
>   File "/data/share/apps/deepmd/2.2/bin/dp", line 8, in <module>
>     sys.exit(main())
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/deepmd/entrypoints/main.py", line 641, in main
>     compress(**dict_args)
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/deepmd/entrypoints/compress.py", line 150, in compress
>     train(
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/deepmd/entrypoints/train.py", line 164, in train
>     _do_work(jdata, run_opt, is_compress)
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/deepmd/entrypoints/train.py", line 283, in _do_work
>     model.save_compressed()
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/deepmd/train/trainer.py", line 1134, in save_compressed
>     self.saver.save(self.sess, os.path.join(os.getcwd(), self.save_ckpt))
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/tensorflow/python/training/saver.py", line 1324, in save
>     self.export_meta_graph(
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/tensorflow/python/training/saver.py", line 1373, in export_meta_graph
>     graph_def=ops.get_default_graph().as_graph_def(add_shapes=True),
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/tensorflow/python/framework/ops.py", line 3568, in as_graph_def
>     result, _ = self._as_graph_def(from_version, add_shapes)
>   File "/data/share/apps/deepmd/2.2/lib/python3.10/site-packages/tensorflow/python/framework/ops.py", line 3482, in _as_graph_def
>     graph.ParseFromString(compat.as_bytes(data))
> google.protobuf.message.DecodeError: Error parsing message with type 'tensorflow.GraphDef'

It seems that TensorFlow version can not match the graph.So in summary:I trained a model with deepmd/2.0-cuda11.3 and got deprecation warnings about graph_util.extract_sub_graph in train.log,When compressing the model with various deepmd versions, I always got initialization error and protobuf decode error.Thanks for your help!

**Comment by njzjz** [comment]
See #787
